gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/ddl/resolver/GfxdPartitionByExpressionResolver.java

```diff
@@ -462,7 +466,13 @@ private Serializable calcRoutingObject(DataValueDescriptor dvd,
     }
     else {
       // case of partitioning by general expression
-      return invokeExpressionEvaluator(dvd, null, lcc);
+      if(snappyStore){
+        return Integer.valueOf(Misc.getHashCodeSnappy(dvd));
+      }else{
+        return invokeExpressionEvaluator(dvd, null, lcc);
+      }
```

Comment:
I don't think this case of general PARTITION BY (expression) should be changed to mean something else (while accepting arbitrary expressions nevertheless). We should change the normal case of PARTITION BY COLUMN () instead when isSnappyStore is true.